heroku/cli
# refactor(pg-v5): Move command 'pg:locks' to CLI

## Description

Moves the `pg:locks` command to CLI, converting to OClif Core and TS.

⚠️ This command has no unit tests ⚠️ 

## Test instructions

You can test by pulling this branch, installing dependencies with `yarn` and building with `yarn build`.

Then you will need to generate a lock to see something in the output. I've tested by opening an interactive `psql` session to the test app's primary database using `heroku pg:psql` on a secondary terminal window, creating a table with:

```sql
CREATE TABLE tests (
  attr1 text,
  attr2 integer,
  attr3 timestamp
);
```

then creating a lock by issuing:

```sql
BEGIN TRANSACTION;
LOCK TABLE tests IN ACCESS EXCLUSIVE MODE;
```

and finally you can running the following commands to test:

```
./bin/run pg:locks --help                  # always test that help looks ok.
./bin/run pg:locks -a mytest-app           # shows locks on the table.
./bin/run pg:locks -a mytest-app -t        # shows locks on the table truncating query string.
```

Compare outputs against the current CLI.

## SOC2 Compliance
[GUS Work Item](https://gus.lightning.force.com/a07EE00001bCw3oYAC) (Salesforce Internal)
